### DISCUSSION TITLE
[RFC]: Strict page handling and routing

### DISCUSSION THREAD

**Discussion by malekjaroslav** [blitz-js/blitz#2789] [Ideas] [answered]
Currently we are able to put the folder `pages` inside any folder in the project and it gets merged with the main `pages` folder in the root. I think this is very hard to reason about. I've spent a lot of time deciding where to put it because it's very ambiguous. 

I suggest it would make life easier for everyone if Blitz offered only one way to handle routing. 

What do you think?

**Comment by prisis** [comment]
Hey :) @malekjaroslav, i think this 2 discussions are show you why you can create pages in different folders https://github.com/blitz-js/blitz/discussions/2631 and https://github.com/blitz-js/blitz/discussions/1694

And for me im using it to create blitz recipes, that have everything in on folder, components, routes, config etc.

**Comment by flybayer** [comment, marked as answer]
You are more than welcome to keep all your pages in a single `pages` folder. In fact that's what we do with `blitz generate`. 

Also remember you have the `blitz routes` command which is helpful to see all the routes in your application. :)
